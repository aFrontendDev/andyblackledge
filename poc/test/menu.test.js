import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import renderer from 'react-test-renderer';
import Menu from '../app/components/menu';

const fakeFunc = () => {};

it('should render the menu', () => {
  const tree = renderer.create(
    <Menu onMenuAction={fakeFunc} />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

it('should render the menu with menuIn prop set', () => {
  const tree = renderer.create(
    <Menu onMenuAction={fakeFunc} menuIsIn={true} />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

it('should simulate menu close btn press', () => {
  const component = shallow(
    <Menu onMenuAction={fakeFunc} />
  );

  expect(toJson(component)).toMatchSnapshot();

  component.find('.menu__close-btn').simulate('click');
  
  expect(toJson(component)).toMatchSnapshot();
});