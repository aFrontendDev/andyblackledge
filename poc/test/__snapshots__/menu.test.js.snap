// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the menu 1`] = `
<div
  className="menu "
>
  <button
    className="btn btn--unstyled menu__close-btn"
    onClick={[Function]}
  >
    Close
  </button>
  <ul
    className="nav"
  >
    <li>
      <a
        href="/"
      >
        Home
      </a>
    </li>
    <li>
      <a
        href="/test"
      >
        Test
      </a>
    </li>
    <li>
      <a
        href="/dynamic/123"
      >
        Dynamic Page
      </a>
    </li>
  </ul>
</div>
`;

exports[`should render the menu with menuIn prop set 1`] = `
<div
  className="menu menu--in"
>
  <button
    className="btn btn--unstyled menu__close-btn"
    onClick={[Function]}
  >
    Close
  </button>
  <ul
    className="nav"
  >
    <li>
      <a
        href="/"
      >
        Home
      </a>
    </li>
    <li>
      <a
        href="/test"
      >
        Test
      </a>
    </li>
    <li>
      <a
        href="/dynamic/123"
      >
        Dynamic Page
      </a>
    </li>
  </ul>
</div>
`;

exports[`should simulate menu close btn press 1`] = `
<div
  className="menu "
>
  <button
    className="btn btn--unstyled menu__close-btn"
    onClick={[Function]}
  >
    Close
  </button>
  <ul
    className="nav"
  >
    <li>
      <a
        href="/"
      >
        Home
      </a>
    </li>
    <li>
      <a
        href="/test"
      >
        Test
      </a>
    </li>
    <li>
      <a
        href="/dynamic/123"
      >
        Dynamic Page
      </a>
    </li>
  </ul>
</div>
`;

exports[`should simulate menu close btn press 2`] = `
<div
  className="menu "
>
  <button
    className="btn btn--unstyled menu__close-btn"
    onClick={[Function]}
  >
    Close
  </button>
  <ul
    className="nav"
  >
    <li>
      <a
        href="/"
      >
        Home
      </a>
    </li>
    <li>
      <a
        href="/test"
      >
        Test
      </a>
    </li>
    <li>
      <a
        href="/dynamic/123"
      >
        Dynamic Page
      </a>
    </li>
  </ul>
</div>
`;
